### PR TITLE
[Jira] Improve the performance of connector

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## PR Content
This PR contains the changes that is improving the connector performance. After updating the code, the connector code improves the performance a way better. Currently, the implementation of fetching issues is sequential and so making it concurrent.

#### Performance Stats
- **before:** ~1 hour 25 mins (50,200 documents with 10 GiB of extractable content)
   - Perf8 Report: https://drive.google.com/file/d/1-R2--jACyIEBrmRN9CrabhfRUXms94d9/view?usp=share_link
- **after:** ~20 mins (45,450 documents with 10 GiB of extractable content)
   - Perf8 Report: https://drive.google.com/file/d/1wdzbA3KRwF49nJ1TMR4mnAAGO3FU-dsd/view?usp=share_link

Please refer this performance stats doc for more details: https://docs.google.com/document/d/1Yo86zQg4ScLAoX6d34lnWoiDQfow7-H7xF3qoH6qisQ

#### Pre-Review Checklist
- [x] this PR has a meaningful title
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally

### CODE REVIEW

**Review by seanstory — DISMISSED** [review]
lgtm

File: connectors/sources/jira.py [hunk 1]
```diff
@@ -272,7 +272,7 @@ def get_default_configuration(cls):
                 "type": "int",
             },
             "concurrent_downloads": {
-                "value": 50,
+                "value": 100,
```
[the remark this answers is not in the document]
Comment (moxarth-elastic) [reply connectors/sources/jira.py:275]: We've got a comment from Artem to make the default values inline. That's why I haven't made it constant. https://github.com/elastic/connectors-python/pull/527#discussion_r1114080849

**Comment by artem-shelkovnikov** [comment]
Hi @moxarth-elastic! Can you resolve conflicts in the PR so that we could merge it?

File: connectors/sources/jira.py [hunk 2]
```diff
@@ -466,27 +481,46 @@ async def _get_projects(self):
...
-                            "_timestamp": timestamp,
-                            "Type": "Project",
-                            "Project": project,
-                        }
+                        await self._put_projects(project=project, timestamp=timestamp)
+            await self.queue.put("FINISHED")  # pyright: ignore
         except Exception as exception:
             logger.warning(f"Skipping data for type: {PROJECT}. Error: {exception}")
 
+    async def _get_specific_issue(self, issue):
```
Comment (artem-shelkovnikov) [inline connectors/sources/jira.py:496]:
```suggestion
    async def _get_specific_issue(self, issue):
```

Should it be a `_put_issue` similar to `_put_projects`?

**Comment by tarekziade** [comment]
Also, related, we can remove some memory usage in https://github.com/elastic/connectors-python/issues/684

**Comment by moxarth-elastic** [comment]
buildkite test this

**Comment by tarekziade** [comment]
> @tarekziade do you think the memory growth is acceptable in this case then?

yes as long as we have knobs to reduce it -- it's a tradeoff . more speed vs less memory usage

**Review by artem-shelkovnikov — APPROVED** [review]

**Pull request merged**